opus/application/static_media/js/opus.js

```diff
@@ -135,11 +139,37 @@ var opus = {
 
         } else {
             // selections in the url hash is different from opus.last_selections
-            // reset the pages: {"gallery":1, "data":1, "cart_gallery":1, "cart_data":1 };
+            // reset the pages:
             opus.prefs.page = default_pages;
 
-            // and reset the query:
-            o_browse.resetQuery();
+            // create an object from selections to compare with opus.selections
+            let modifiedSelections = {};
+            $.each(Object.keys(selections), function(idx, slug) {
+                // qtype is not updated in opus.selections
+                if (!slug.match(/qtype/)) {
+                    modifiedSelections[slug] = selections[slug][0].replace("+", " ").split(",");
+                }
+            });
+
+            // debug print out
+            // console.log("=== selections ===")
+            // console.log(selections);
+            // console.log("=== modifiedSelections ===")
+            // console.log(modifiedSelections);
+            // console.log("=== opus.selections ===");
+            // console.log(opus.selections);
+            // console.log("Are modifiedSelections and opus.selections the same ?");
+            // console.log(opus.checkIfSelectionsObjectsAreTheSame(modifiedSelections, opus.selections));
+
+            // if data in selections !== data in opus.selections, it means selections are modified manually in url, reload the page (modified url in url bar and hit enter)
+            if (!opus.checkIfSelectionsObjectsAreTheSame(modifiedSelections, opus.selections)) {
```

Comment:
Generally when a function returns a True/False value, you don't want to call it something that starts with a verb. "Check If" means perform some kind of action, but doesn't tell you that it's going to return a True/False value based on the result. Instead, use a wording like "(!opus.selectionsObjectsAreTheSame...)". Then it makes sense when you read it as "if not selections objects are the same".

Reply:
=== tests for the exact same object, not the same contents. It’s obvious from the code that the objects will never be the same object. But they could have the same contents. As discussed in slack the problem is that opus allows both “planet=“ and (nothing) for an empty search string. This is supposed to handle those cases, which are unique to opus search selections.